packages/@tinacms/cli/src/cmds/forestry-migrate/util/errorSingleton.ts

```diff
@@ -0,0 +1,80 @@
+/**
+Copyright 2021 Forestry.io Holdings, Inc.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import { logger } from '../../../logger'
+import { dangerText, linkText } from '../../../utils/theme'
+
+interface NameError {
+  name: string
+  template: string
+  newName: string
+}
+
+/**
+ * The Singleton class defines the `getInstance` method that lets clients access
+ * the unique singleton instance.
+ */
+export class ErrorSingleton {
+  private static instance: ErrorSingleton
+
+  /**
+   * The Singleton's constructor should always be private to prevent direct
+   * construction calls with the `new` operator.
+   */
+  private constructor() {}
+
+  /**
+   * The static method that controls the access to the singleton instance.
+   *
+   * This implementation let you subclass the Singleton class while keeping
+   * just one instance of each subclass around.
+   */
+  public static getInstance(): ErrorSingleton {
+    if (!ErrorSingleton.instance) {
+      ErrorSingleton.instance = new ErrorSingleton()
+      ErrorSingleton.instance.allErrorNames = []
+    }
+
+    return ErrorSingleton.instance
+  }
+
+  private allErrorNames: NameError[]
+
+  public addErrorName(error: NameError) {
+    this.allErrorNames.push(error)
+  }
+  public printErrorNames() {
```

Comment:
```suggestion
  public printNameErrors() {
```
(You'll have to update the function calls as well)